coalib/bears/Bear.py

```diff
@@ -423,9 +423,14 @@ def download_cached_file(self, url, filename):
                   .format(filename=filename, bearname=self.name, url=url))
 
         response = requests.get(url, stream=True, timeout=20)
+        response.raise_for_status()
+
         with open(filename, 'wb') as file:
             for chunk in response.iter_content(125):
                 file.write(chunk)
+
+        response.raise_for_status()
```

Comment:
The first one uses the initial status code, raising an exception before opening the file.  Obviously not a problem.

I added the second one as super defensive programming (almost as a TODO item for me and the reviewer, so I am glad you questioned it).  If there is any chance of a new response during the `iter_content`, a new response code will occur, and maybe the response code is not the same as the original one.

One aspect that may trigger a new request is retries  -- surely a new request is needed to retry the fetch.  I havent thought of any other scenario where the response status code might change during the content fetch phase.

And I can leave this second (and probably useles?) `raise_for_status` out of this PR, and tackle the retries problem in the separate bug https://github.com/coala/coala/issues/3804.